clippy2021.md

```diff
@@ -0,0 +1,178 @@
+addressed later in this document.
+
+[#6429](https://github.com/rust-lang/rust-clippy/issues/6429)
+
+#### Clippy book
+
+We should consider writing a Clippy book, like other projects have. This will
+then be the central place where everything you need to know to use Clippy is
+collected in a structured way.
+
+[#6011](https://github.com/rust-lang/rust-clippy/issues/6011)
+
+### Reliability
+
+We should make sure to make Clippy more reliable and stable for our users
+
+#### FP rate
+
+In the worst case, new lints are only available in nightly for 2 weeks, before
+hitting beta and ultimately stable. This and the fact that fewer people use
+nightly Rust nowadays makes it more more probably that a lint with many FPs hits
+stable. This leads to annoyed users, that will disable these new lints in the
+best case and to more annoyed users, that will stop using Clippy in the worst.
+We should think of a process to prevent this from happening. This will also be
+addressed later in this issue.
+
+[#6429](https://github.com/rust-lang/rust-clippy/issues/6429)
+
+## Internal
+
+(The end of) 2020 has shown, that Clippy has to think about the available
+resources, especially regarding management and maintenance of the project. This
+section address issues affecting team members and contributors.
+
+### Management
+
+In 2020 we got over 1000 open issues with now between 25-35 open PRs. This is
+mostly because of the lack of resources in the team. According to the planned
+roadmap of the Rust project, we should improve our processes and look for more
+eyes and hands to help reviewing and triaging PRs and issues.
+
+#### Clear expectations for team members
+
+We will have to write this out according to the planned Rust 2021 roadmap
+anyway. This should not put more pressure on the team members, but rather help
+them and interested folks to know what our expectations are.
+
+#### More maintainer
+
+More people means less work for each individual. We should work out a process
+how onboarding of new folks can work and look out for people that are interested
+joining and taking up responsibility.
+
+#### (Bi-)weekly meetings
+
+Other teams have regular meetings. Clippy is big enough, that it might be worth
+it to also do them. Especially if we want/get more people joining the team, this
+can be important for sync-ups. I for example can't keep track of all the new
+lints and features that are implemented in Clippy, because I stopped looking at
+every PR a while ago. And no one should have to do this to be up-to-date.
+
+### Development
+
+We can always improve the developer and contributor experience with Clippy and
+simultaneously make the life of the team members easier.
+
+#### Process for new lints
+
+As already mentioned above, classifying new lints gets quite hard, because the
+probability of a buggy lint getting into stable is quite high. We should work
+out a process (and maybe tooling) to make sure we test lints, before releasing
+them into the wild.
```

Comment:
I agree, I think we should be much more careful before enabling new lints by default. We definitely need a tool or good instructions to make testing new lints on _real-world_ code easier. From my experience, the harder it is to test something, the less likely it is to be tested properly at all. A quick feedback loop is very important, imo.

(Also see [this comment here](https://github.com/rust-lang/rust-clippy/issues/6429#issuecomment-741153345) about cppcheck testing)